### DISCUSSION TITLE
WPF: `IChromiumWebBrowserBase.Find()` questions

### DISCUSSION THREAD

**Discussion by chucker** [cefsharp/CefSharp#4721] [General]
I was able to implement a basic find panel that highlights matches. However:

* Edge (and presumably other Chromium browsers) has a notion of the 'primary' search result, and highlights that in orange instead of yellow. I'm unsure if this is missing in CEF, or if I need to first implement `IFindHandler` to indicate which result I wish to highlight was 'primary'. Without this, I don't really see the point of the `forward` and `findNext` parameters; they don't appear to make a user-visible difference.
* `IFindHandler.OnFindResult` gives me a `selectionRect`. It would be nice to jump to that and/or scroll it into view if not currently visible. However, I can't find any method that will let me scroll that rect into view. The closest I can find are various JS-based approaches (e.g., https://github.com/cefsharp/CefSharp/discussions/4003). Is that really the best approach?

**Comment by amaitland** [comment]
I've not needed to use the `Find API` for some time, so it's possible that it's not as intended anymore.

Basically `CefSharp` just provides a wrapper around the `CEF API`.

https://cef-builds.spotifycdn.com/docs/122.1/classCefBrowserHost.html#a7b7359016e0b6965559e9af48eaf636b

There's an example of using the API in the CEF sample application. https://github.com/chromiumembedded/cef/blob/2c5dd120c74a2dc61f07b9043ed26ccb977298c3/tests/cefclient/browser/root_window_win.cc#L905

> * Edge (and presumably other Chromium browsers) has a notion of the 'primary' search result, and highlights that in orange instead of yellow. I'm unsure if this is missing in CEF, or if I need to first implement `IFindHandler` to indicate which result I wish to highlight was 'primary'. Without this, I don't really see the point of the `forward` and `findNext` parameters; they don't appear to make a user-visible difference.

From memory it should have a similar behaviour to `Chromium`.

I'd suggest testing with the `CEF Sample Application (cefclient)` and seeing how it behaves.

Download and extract https://cef-builds.spotifycdn.com/cef_binary_122.1.7%2Bgdd187af%2Bchromium-122.0.6261.29_windows64_client.tar.bz2

Launch cefclient with the following command line args (this is very important)

```
 cefclient.exe --multi-threaded-message-loop --no-sandbox --off-screen-rendering-enabled --enable-gpu
```
